lib/sherlock_homes/trulia.rb

```diff
@@ -64,27 +64,62 @@ def parse
 
     class SummaryParser < AbstractParser
       def parse
-        result = []
+        result = Hash.new
         @data.css('li').each do |li|
-          result << li.text.strip
+          text = li.text.strip
+          next if text.eql? "Edit Home Facts"
+          result.merge! BulletItemParser.parse(text)
         end
-        result.delete("Edit Home Facts")
         result
       end
     end
 
     class PublicRecordsParser < AbstractParser
       def parse
-        result = []
+        result = Hash.new
         @data.css('li').each do |group|
           group.css('ul > li').each do |li|
-            result << li.text.strip
+            result.merge! BulletItemParser.parse(li.text.strip)
           end
         end
         result
       end
     end
 
```

Comment:
@jeffdeville Thanks you for the suggestion, I'm going to define those parsers as lambdas. Sometimes I still suffer a strict OOP mindset, whereas I should think more functional!